src/pages/link/link.js

```diff
@@ -0,0 +1,62 @@
+const router = require('express').Router()
+const { sqlLoader } = require('@prairielearn/prairielib')
+const asyncErrorHandler = require('../../asyncErrorHandler')
+const dbDriver = require('../../dbDriver')
+const checks = require('../../auth/checks')
+
+const sql = sqlLoader.loadSqlEquiv(__filename)
+
+router.get(
+  '/',
+  asyncErrorHandler(async (req, res, _next) => {
+    if (!(await checks.isLoggedIn(req))) {
+      res.redirect('/login') // TODO: redirect back after login
+      return
+    }
+
+    const params = {
+      email: req.user.email,
+    }
+
+    const results = await dbDriver.asyncQuery(
+      sql.get_linked_students_with_info,
+      params
+    )
+
+    res.locals.linked_course_data = results.rows
+
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals)
+  })
+)
+
+router.post(
+  '/',
+  asyncErrorHandler(async (req, res, _next) => {
+    if (!(await checks.isLoggedIn(req))) {
+      res.sendStatus(403)
+      return
+    }
+
+    const uin = Number.parseInt(req.body.uin, 10)
```

Comment:
**Warning:** `parseInt` acts like`strol` in C.

Example
```js
function roughScale(x, base) {
  var parsed = Number.parseInt(x, base);
  if (Number.isNaN(parsed)) {
    return 0;
  }
  return parsed ;
}

console.log(roughScale(' 0xF', 16));
// expected output: 15

console.log(roughScale('321 troll', 10));
// expected output: 321
```

Reply:
Actually, guessing you use `atoi` more but still same behavior.

Reply:
Hmmmm, you may actually want that behavior for handling the swipers if you actually care. Personally, I think you should restrict the string length and then parse as a universal solution (not sure what garbage the swiper puts after the UIN, like for example another number).

Reply:
not sure how much we care about this; We would also want/need to change all the other number inputs around the app to whatever method accepts if a real number, not just prefix

Reply:
Ya, I agree. I didn't give this much thought before. Could make a PR to sweep through and check for this for inputs of type `number` passed to the `res.body`.